File: CS/Models/CustomReportResolver.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Web.WebDocumentViewer.Native.Services;

namespace MVCViewerWidgetDemo.Models {
    public class CustomReportResolver : WebDocumentViewerReportResolver {
        public override XtraReport Resolve(string reportTypeName) {
            var report = base.Resolve(reportTypeName);
            if(report == null) {
                switch(reportTypeName) {
                    case "MVCViewerWidgetDemo.Reports.OrdersReport":
                        report = new Reports.OrdersReport();
                        break;
                    case "MVCViewerWidgetDemo.Reports.ProductsReport":
                        report = new Reports.ProductsReport();
                        break;
                }
            }
            return report;
        }
    }
}

File: VB/Models/CustomReportResolver.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Web.WebDocumentViewer.Native.Services

Namespace MVCViewerWidgetDemo.Models
    Public Class CustomReportResolver
        Inherits WebDocumentViewerReportResolver

        Public Overrides Function Resolve(ByVal reportTypeName As String) As XtraReport
            Dim report = MyBase.Resolve(reportTypeName)
            If report Is Nothing Then
                Select Case reportTypeName
                    Case "MVCViewerWidgetDemo.Reports.OrdersReport"
                        report = New Reports.OrdersReport()
                    Case "MVCViewerWidgetDemo.Reports.ProductsReport"
                        report = New Reports.ProductsReport()
                End Select
            End If
            Return report
        End Function
    End Class
End Namespace